src/Controller/Controller.php

```diff
@@ -227,8 +227,9 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      * @param \Cake\Network\Response|null $response Response object for this controller.
      * @param string|null $name Override the name useful in testing when using mocks.
      * @param \Cake\Event\EventManager|null $eventManager The event manager. Defaults to a new instance.
+     * @param \Cake\Controller\ComponentRegistry|null $components The component registry. Defaults to a new instance.
      */
-    public function __construct(Request $request = null, Response $response = null, $name = null, $eventManager = null)
+    public function __construct(Request $request = null, Response $response = null, $name = null, $eventManager = null, $components = null)
     {
```

Comment:
@markstory well, request and response was there in 2.x, so i figure we can make it the same.  I don't mind getting them all into $config though.

@korotovsky to make it uniform like other classes do, eg: [Table::__construct](https://github.com/cakephp/cakephp/blob/master/src/ORM/Table.php#L233-L235)